### DISCUSSION TITLE
Concatenating KroneckerProduckLinearOperator objects

### DISCUSSION THREAD

**Discussion by AndreaBraschi** [cornellius-gp/linear_operator#77] [Q&A]
Hi all,

I am trying to concatenate 2 `KroneckerProductLinearOperator` objects of shapes `(100, 100)` and `(50, 50)` in a way that the resulting `LinearOperator` is `shape(150, 150)`. i was wondering if there is a way of doing it without having to evaluate the `LinearOperator` objects?

Many thanks in advance.

Andrea

**Comment by Balandat** [comment]
What do you mean by concatenating? In the standard meaning of concatenation these the two have to share at least one dimension, which isn't the case here. Do you maybe mean that you want to build a block diagonal operator out of these? `BlockDiagLinearOperator` can help with that.

**Reply by AndreaBraschi** [reply]
Hi @Balandat,

many thanks for getting back to me. 

Yes, my idea was to create a block-diagonal structure with square matrices with different size.
I thought the `BlockDiagonalLinearOperator` only worked with square matrices with homogenous size. I'll have a go trying that class.

Andrea

**Reply by Balandat** [reply]
Ah yeah I guess you're right that's not supported at this point. See related discussion here: https://github.com/cornellius-gp/linear_operator/discussions/73
